### ISSUE TITLE
why would you update item_embeddings with the linear transformation of o. 

### ISSUE THREAD

**Issue by huanzhao595** [hwwang55/RippleNet#29]
`self.transform_matrix = nn.Linear(self.dim, self.dim, bias=False)

item_embeddings = self._update_item_embedding(item_embeddings, o)
            
 item_embeddings = self.transform_matrix(item_embeddings + o)
`
As the codes above show, vector o can be seen as the 1-order response of user u’s click history Vu with respect to the candidate item.  Item_embeddings can be seen as the candidate item embeddings. 

I have a question, why would you update item_embeddings with the linear transformation of o. 

To be honest, I drop this code and the precision decreased dramatically, the highest precision is close to  50% on Movielens. So what's the purpose of this linear transformation.

PS, I didn't find any theory in the paper to explain this code. Please tell me why.

**Comment by zhoupengfeigjqh** [comment]
+1
